{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table style=\"border: 2px solid white;\">\n",
       "<tr>\n",
       "<td style=\"vertical-align: top; border: 0px solid white\">\n",
       "<h3 style=\"text-align: left;\">Client</h3>\n",
       "<ul style=\"text-align: left; list-style: none; margin: 0; padding: 0;\">\n",
       "  <li><b>Scheduler: </b>tcp://198.202.103.227:35585</li>\n",
       "  <li><b>Dashboard: </b><a href='http://198.202.103.227:8787/status' target='_blank'>http://198.202.103.227:8787/status</a></li>\n",
       "</ul>\n",
       "</td>\n",
       "<td style=\"vertical-align: top; border: 0px solid white\">\n",
       "<h3 style=\"text-align: left;\">Cluster</h3>\n",
       "<ul style=\"text-align: left; list-style:none; margin: 0; padding: 0;\">\n",
       "  <li><b>Workers: </b>0</li>\n",
       "  <li><b>Cores: </b>0</li>\n",
       "  <li><b>Memory: </b>0 B</li>\n",
       "</ul>\n",
       "</td>\n",
       "</tr>\n",
       "</table>"
      ],
      "text/plain": [
       "<Client: 'tcp://198.202.103.227:35585' processes=0 threads=0, memory=0 B>"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import torch\n",
    "import os\n",
    "import sys\n",
    "sys.path.append('../')\n",
    "if 'DISPLAY' in os.environ:\n",
    "    del os.environ['DISPLAY']\n",
    "\n",
    "import numpy as np\n",
    "import dill\n",
    "import matplotlib.pyplot as plt\n",
    "import joblib\n",
    "from distributed import Client\n",
    "from dask_jobqueue import SLURMCluster\n",
    "import dask\n",
    "from joblib import Parallel, delayed\n",
    "import time\n",
    "import random\n",
    "import pandas as pd\n",
    "import hnn_core\n",
    "from hnn_core import simulate_dipole, Network, read_params, JoblibBackend, MPIBackend\n",
    "import sbi.utils as utils\n",
    "import datetime\n",
    "import dask.bag as db\n",
    "#from sbi_functions import HNNSimulator, run_simulator\n",
    "\n",
    "\n",
    "cluster = SLURMCluster(cores = 100,\n",
    "                       processes=100,\n",
    "                       queue='compute',\n",
    "                       memory=\"100GB\",\n",
    "                       walltime=\"24:00:00\",\n",
    "                       job_extra=['-A csd403', '--nodes=1']\n",
    ")\n",
    "#'--cpus-per-task=1'\n",
    "client = Client(cluster)\n",
    "client"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "class HNNSimulator:\n",
    "    def __init__(self, params_fname, prior_dict):\n",
    "        if 'DISPLAY' in os.environ:\n",
    "            del os.environ['DISPLAY']\n",
    "            \n",
    "        import hnn_core\n",
    "        from hnn_core import simulate_dipole, Network, read_params, JoblibBackend, MPIBackend\n",
    "        self.params = read_params(params_fname)\n",
    "        #self.params['tstop'] = 100\n",
    "        self.param_names = list(prior_dict.keys())\n",
    "\n",
    "    def __call__(self, new_param_values):\n",
    "        new_params = dict(zip(self.param_names, new_param_values.detach().cpu().numpy()))\n",
    "        self.params.update(new_params)\n",
    "\n",
    "        net = Network(self.params)\n",
    "        with JoblibBackend(n_jobs=1):\n",
    "            dpl = simulate_dipole(net, n_trials=1)\n",
    "\n",
    "        summstats = dpl[0].data['agg']\n",
    "        spike_times = net.cell_response.spike_times\n",
    "        spike_gids = net.cell_response.spike_gids\n",
    "        spike_types = net.cell_response.spike_types\n",
    "        return summstats, spike_times, spike_gids, spike_types\n",
    "\n",
    "\n",
    "#sbi_simulator, sbi_prior = prepare_for_sbi(hnn_simulator, prior)\n",
    "#params = read_params(params_fname)\n",
    "def run_simulator(theta, params_fname, prior_dict, sim_idx):\n",
    "    hnn_simulator = HNNSimulator(params_fname,prior_dict)\n",
    "    dpl, spike_times, spike_gids, spike_types = hnn_simulator(theta)\n",
    "    return (dpl, spike_times, spike_gids, spike_types)\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "num_simulations = 10000\n",
    "time_stamp = datetime.datetime.now().strftime(\"%m%d%Y_%H%M%S\")\n",
    "\n",
    "params_fname = '../../data/ERP/ERPYes.param'\n",
    "save_suffix = 'ERP_Yes_t10000' + '_' + time_stamp\n",
    "save_path = '../../data/ERP/prerun_simulations/' + save_suffix  + '/'\n",
    "\n",
    "prior_dict = {'gbar_L2Pyr_L2Pyr_ampa':(0, 0.01),\n",
    "'gbar_L2Pyr_L2Pyr_nmda':(0, 0.01), \n",
    "'gbar_L2Basket_L2Pyr_gabaa':(0, 0.1),\n",
    "'gbar_L2Basket_L2Pyr_gabab':(0, 0.1),\n",
    "'gbar_L2Pyr_L5Pyr':(0, 0.01),\n",
    "'gbar_L2Basket_L5Pyr':(0, 0.1),\n",
    "'gbar_L5Pyr_L5Pyr_ampa':(0, 0.01),\n",
    "'gbar_L5Pyr_L5Pyr_nmda':(0, 0.01),\n",
    "'gbar_L5Basket_L5Pyr_gabaa':(0, 0.1),\n",
    "'gbar_L5Basket_L5Pyr_gabab':(0, 0.1),\n",
    "'gbar_L2Pyr_L2Basket':(0, 0.01),\n",
    "'gbar_L2Basket_L2Basket':(0, 0.1),\n",
    "'gbar_L2Pyr_L5Basket':(0, 0.01),\n",
    "'gbar_L5Pyr_L5Basket':(0, 0.01),\n",
    "'gbar_L5Basket_L5Basket':(0, 0.1)}\n",
    "\n",
    "param_low = [float(item[0]) for key, item in prior_dict.items()]\n",
    "param_high = [float(item[1]) for key, item in prior_dict.items()]\n",
    "prior = utils.BoxUniform(low=torch.tensor(param_low), high=torch.tensor(param_high))\n",
    "\n",
    "theta_samples = prior.sample((num_simulations,))\n",
    "\n",
    "def dill_save(save_object, save_prefix, save_suffix, save_path, extension='.pkl'):\n",
    "    save_file = open(save_path + save_prefix + '_' + save_suffix + extension, 'wb')\n",
    "    dill.dump(save_object, save_file)\n",
    "    save_file.close()\n",
    "\n",
    "os.mkdir(save_path)\n",
    "os.mkdir(save_path + 'data/')\n",
    "dill_save(params_fname, 'params_fname', save_suffix, save_path)\n",
    "dill_save(prior, 'prior', save_suffix, save_path)\n",
    "dill_save(prior_dict, 'prior_dict', save_suffix, save_path)\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-1aa5f726-a490-444c-80b1-c87508c6a48f'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-27ff787d-dd81-4271-b0df-a87bfc32a4ea'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-1c2e39bc-e63c-4a93-9913-b801c33a3de4'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-07eb882f-ab98-4a0e-b065-a9dbfb47b5f2'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-11477a89-9e9b-4d60-8b0e-0e1b445a98c5'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-365cb7e2-c2e0-453e-99c5-6d6ed37eb05a'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-1fb4ebc6-3fd8-4e07-bad3-60772b83498b'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-0ea0091d-65c1-4bf2-b3a4-00ee2786ef1e'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-056767f8-2ea3-4dc3-9137-53690006e1ea'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-0990b8ae-3658-4c45-a18d-7624be07c6df'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-17efe9c7-a1da-4700-9180-46f81ba61ae0'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-020bda03-d7f7-4016-8160-a4345a2d6219'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-05cd0c01-fb72-4aee-ba95-c5888818eecb'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-14ad837b-cfbf-4467-b722-e010830c10e9'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-0b75be49-320c-4abd-93ab-fe0a2ab477df'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-06a8a786-cd97-4a50-bc53-13658b0656a1'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-011272b9-2004-492c-af35-48c6dfaa0e0e'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-04d06b5c-acd4-454a-a66f-60df3dce7dd3'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-026e137c-10b5-41ff-b037-dfccb6d37e11'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-30e9a60d-87ea-44a7-bc24-d8daf4d4bade'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-0635114b-fcc4-4dbc-9250-1237bc51ded9'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-46b52c78-9475-4741-9c8c-8782447af378'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-13c03677-398d-4816-b42d-0ad053dd0904'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-de72a249-526c-4ee4-bf77-9dbb68bf45a8'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-0547296b-c015-4a62-b54e-a093e4226e43'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-f4561748-5ff9-49d4-99ca-871f93ee867b'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-a08c3916-febf-4d30-a138-233f087d742c'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-5e53778a-354b-403e-a50d-a205963c81ad'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-3645c498-8d04-4a1d-91f4-af834c501299'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-455483aa-724a-4df9-804d-553afec8ffb8'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-253cfe68-1090-45c5-8129-b182f3b0f272'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-3fd92676-2fca-4c8b-a827-0fcb79bedbd5'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-e5383af3-5582-440e-8201-ddcd883774a8'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-2ece6e5f-dd2d-4c46-9837-6376432f29be'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-e1ad32d9-6110-4287-ab67-87ae9738c9b0'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-2ece6e5f-dd2d-4c46-9837-6376432f29be'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-455483aa-724a-4df9-804d-553afec8ffb8'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-e5383af3-5582-440e-8201-ddcd883774a8'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-253cfe68-1090-45c5-8129-b182f3b0f272'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-2ece6e5f-dd2d-4c46-9837-6376432f29be'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-a08c3916-febf-4d30-a138-233f087d742c'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-88c922d5-3434-4329-907f-ad6a442381d8'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-e1ad32d9-6110-4287-ab67-87ae9738c9b0'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-88c922d5-3434-4329-907f-ad6a442381d8'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-5e53778a-354b-403e-a50d-a205963c81ad'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-f4561748-5ff9-49d4-99ca-871f93ee867b'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-3645c498-8d04-4a1d-91f4-af834c501299'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-3fd92676-2fca-4c8b-a827-0fcb79bedbd5'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-de72a249-526c-4ee4-bf77-9dbb68bf45a8'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-88c922d5-3434-4329-907f-ad6a442381d8'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-88c922d5-3434-4329-907f-ad6a442381d8'\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-5-0516d2269eae>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     25\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnum_simulations\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstep_size\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     26\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 27\u001b[0;31m     \u001b[0mbatch\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mi\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mstep_size\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mtheta_samples\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparams_fname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mprior_dict\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-5-0516d2269eae>\u001b[0m in \u001b[0;36mbatch\u001b[0;34m(seq, theta_samples, params_fname, prior_dict)\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m     \u001b[0mfinal_res\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdask\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcompute\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mres_list\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     11\u001b[0m     \u001b[0mdpl_list\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstack\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mfinal_res\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0midx\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0midx\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/sbi/lib/python3.8/site-packages/dask/base.py\u001b[0m in \u001b[0;36mcompute\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    559\u001b[0m         \u001b[0mpostcomputes\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__dask_postcompute__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    560\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 561\u001b[0;31m     \u001b[0mresults\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mschedule\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdsk\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkeys\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    562\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mrepack\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0ma\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mzip\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresults\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpostcomputes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    563\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/client.py\u001b[0m in \u001b[0;36mget\u001b[0;34m(self, dsk, keys, restrictions, loose_restrictions, resources, sync, asynchronous, direct, retries, priority, fifo_timeout, actors, **kwargs)\u001b[0m\n\u001b[1;32m   2682\u001b[0m                     \u001b[0mshould_rejoin\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2683\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2684\u001b[0;31m                 \u001b[0mresults\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgather\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpacked\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0masynchronous\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0masynchronous\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdirect\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdirect\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2685\u001b[0m             \u001b[0;32mfinally\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2686\u001b[0m                 \u001b[0;32mfor\u001b[0m \u001b[0mf\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mfutures\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/client.py\u001b[0m in \u001b[0;36mgather\u001b[0;34m(self, futures, errors, direct, asynchronous)\u001b[0m\n\u001b[1;32m   1991\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1992\u001b[0m                 \u001b[0mlocal_worker\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1993\u001b[0;31m             return self.sync(\n\u001b[0m\u001b[1;32m   1994\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_gather\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1995\u001b[0m                 \u001b[0mfutures\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/client.py\u001b[0m in \u001b[0;36msync\u001b[0;34m(self, func, asynchronous, callback_timeout, *args, **kwargs)\u001b[0m\n\u001b[1;32m    837\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mfuture\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    838\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 839\u001b[0;31m             return sync(\n\u001b[0m\u001b[1;32m    840\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloop\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcallback_timeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcallback_timeout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    841\u001b[0m             )\n",
      "\u001b[0;32m~/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/utils.py\u001b[0m in \u001b[0;36msync\u001b[0;34m(loop, func, callback_timeout, *args, **kwargs)\u001b[0m\n\u001b[1;32m    335\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    336\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_set\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 337\u001b[0;31m             \u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwait\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m10\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    338\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    339\u001b[0m         \u001b[0mtyp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mexc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtb\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/sbi/lib/python3.8/threading.py\u001b[0m in \u001b[0;36mwait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    556\u001b[0m             \u001b[0msignaled\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_flag\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    557\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0msignaled\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 558\u001b[0;31m                 \u001b[0msignaled\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_cond\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwait\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    559\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0msignaled\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    560\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/sbi/lib/python3.8/threading.py\u001b[0m in \u001b[0;36mwait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    304\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    305\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0mtimeout\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 306\u001b[0;31m                     \u001b[0mgotit\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mwaiter\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0macquire\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    307\u001b[0m                 \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    308\u001b[0m                     \u001b[0mgotit\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mwaiter\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0macquire\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-812e7add-2446-40e7-b569-099d0bd8e355'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-5c94083f-a10e-4ee9-ab8a-4deda35b7f9b'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-5f152ee6-d45e-41a6-a2be-b4b274456a3a'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-3590b8f1-eb52-4840-8e70-39ddcae8e290'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-a9f10c91-0e24-4399-a212-bc247d196156'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-43d8e14e-bb8c-4314-bdf2-1c8715a8ff5e'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-c8d6c07d-81bf-453b-906c-66ed9129117d'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-ce8ba2f7-ba61-415f-95dd-2321da9155de'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-86428ccd-3f40-4ff7-a537-63ed3314bebb'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-df10260e-c581-4b3c-b55d-cd3d167cb57e'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-b4383691-ebc6-481f-8c47-631878087bb9'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-4394b683-63d4-4b0f-89df-9b48e4913396'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-dff642e2-1beb-49b7-8753-9746cabe32ae'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-5db99839-0fcb-47e7-86fa-23fda0f1b53e'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-5c31eb04-973b-4250-86b5-533b9b61b860'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-59f34289-174e-44e4-b4a3-7b8a82c94031'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-e9335fd4-1c04-4e58-8511-b02028e784e9'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-1becd734-6b01-469d-a5b6-2751c8e1ab87'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-9cfdabda-a985-4652-97a8-853a42396f43'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-4a138667-9fab-4c07-b70e-c1f528d33a3e'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-1a3877ef-3e0f-4e72-94cc-20594e062c12'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-cb9a7caf-0ee6-4201-b9e9-71cb8bd8fa9d'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-17bf396e-53cb-46a6-a64c-16c9287f4297'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-12089c3c-d679-4c2f-a445-4a2ac4264ddc'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-e6432cb6-726b-438f-a125-7b00d79f41cd'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-0ebe9250-29e9-496d-b787-5837e8249119'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-85c1f77d-d603-4e81-ba0c-da7db1fa97cb'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-c74657a4-0bdd-4637-9530-a8c4a0d1e8ca'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-195c2cef-1c55-4a02-9908-6c76cdf42491'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-d37c6fc6-7da9-4641-ae77-8760bea57e53'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-e693a807-cb75-4e57-8447-8500658267a5'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-f7cb409d-ede5-4fb8-8f02-5e2aa7968a83'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-4a9f757f-b87a-4c40-8d46-0cfe5655c316'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-d8a9e592-053a-448d-8977-4510ec45b584'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-aece5db9-5048-4025-8f6d-207650d5fe97'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-b7d32f1a-f034-43e6-a9d8-2a36ce60249c'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-5e6abad8-f2b7-4030-bc20-9e716dca8dac'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-179e7924-4267-4995-9b8f-645c83f4881c'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-7fc77d7e-eb0a-4bbe-89b4-ee2ef6874cb2'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-e3f96be4-cfac-420f-88d5-248d233a976d'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-fbf2c1bf-8e3d-4528-9598-accfae18221c'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-ab84ea58-a556-4424-9917-99499449cad0'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-4c2fdd89-e124-4de1-a65d-dfa1de5594e5'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-d029af85-fad1-456d-9dc0-0deeb1a05693'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-4327ecf8-7253-4990-9371-d1fc438a02b5'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-e6677861-f6a2-4652-9a63-905af57a7b40'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-80c41b56-f4b2-48f9-9070-a0b496d1c83f'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-067b80cc-dcc8-4099-8eb2-304a59c41694'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-0e13f7d1-3767-434f-b487-3246d4743a32'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-fb764600-25b2-4272-9aa7-56a665f57abe'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-9fea9c1d-014d-4439-891f-81a8f39df312'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-4534ca88-e8dd-43d7-b94d-5bacce509ccb'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-6615e353-064b-4967-b644-c8a49ab77bf3'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-99dc9a5c-0e27-436f-b9fe-d1e5b1e93671'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-5f3ff9a7-843a-4ae3-8e4c-125ed53d537d'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-67a9d00f-1e1f-4d0f-95c8-063608476451'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-b4636a9f-c8e5-44ef-ac56-dbfc20019b91'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-80921d98-ca1b-4f70-b39e-0b80758ac75b'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-8f8332f3-9029-423a-9cce-bb8de9a45146'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-b4c69679-a512-4522-9b78-1fb5671baf71'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-1fd2803c-c81e-448d-881c-a248ef142eba'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-30c4e73e-a030-4545-9636-d293e3f447d8'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-4cfd04b0-5474-433a-8e85-396b0f2f1e88'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-e5cd4a15-ba86-4072-bb52-d54cfca6fb7e'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-7d1fb9fc-75d3-4ffe-8a4d-076fd3e269d7'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-003769e1-5787-4ea0-a306-938fd5d32cfb'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-ae38d1ea-12db-4f12-bd82-285daf4f42a1'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-cdcefdbc-c992-442b-954a-a9457ab7b8de'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-5db60394-8757-4393-b83c-9605f1b977f8'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-9b369970-5090-4059-9f02-aa5130ba9da0'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-e58d9f6f-7975-4870-87ab-9b20c10deba3'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-b8d6e66b-c8d5-459a-8999-44fb5ef985f3'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-40f7defd-ed44-4ab3-9bed-40378b54f17f'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-be62a72b-b7cd-4fae-ab99-7a34c421372e'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-c34d8e10-f6c1-41f1-b24d-32881e01deb0'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-52355d5c-ebff-46e7-95a1-ccb6924f67b5'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-23b8cd4a-7283-4369-9a73-3bbe02f11709'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-d845089b-e0f5-4b87-a0c6-14dfe8755c20'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-508b729e-77ee-402a-ad06-ae8ced1802e7'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-9e8beca9-9841-48e0-8930-610273925ab4'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-e07811e2-312c-4915-8f37-27d284e4f324'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-f9364814-aafc-482a-a38c-828ba16b933e'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-ce4d4ffe-1d85-4111-a417-ba65b819e49f'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-d04f6b52-a0e4-4b19-8426-156fe1841d01'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-80cf7d15-1977-4d35-b465-036d4abdbe7b'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-42c2f4e3-985b-4fcf-bd40-30454437927c'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-942651c0-d7ce-4bc3-b432-241964f77a92'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-4bf99c3c-f5c4-4254-849d-2e7c5990eb9f'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-62e3c7b2-5a51-4d8c-ae0f-8be3a9b71dea'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-53c5b239-4c22-4202-bb12-a538790768cc'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-4378d24c-ea40-4a32-88fc-f11bffc051dd'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-e506b001-6ec7-422c-ad1d-a8a988081127'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-fb7609c9-cd59-4656-be5d-14a4d02fb264'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-15f13173-394f-429a-a5bb-15d09e2ffa33'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-7d7ba67e-4b51-467c-945d-b1bffb4ad3fe'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-3d69b908-a029-451f-95a9-c75edc7fa4be'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-95516059-1fdd-4624-ad77-09e45cf8a8e0'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-728c0354-1b90-4b9f-8b2c-4fd4253a3227'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-aa3c1fe7-eec8-4bef-9ffd-7460bbc02e59'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-df551b8e-3be1-4ee9-b243-2fbcde7f3882'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-2ce92f04-08b3-4517-b845-4efdac58d538'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-85a89abf-6fc5-4d02-bafa-6917e9635c62'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-25ec4ea1-a3a4-4b57-b11f-8de6c0da6536'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-b6d86554-334c-49e7-a794-4b87af9cd8b4'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-7d9a561c-3d6f-46c0-af46-cb34b67f9f45'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-e19140d9-ee10-40d9-a688-92c557b591d9'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-a1dea7de-91e2-4625-b1a4-2ebfcca419b2'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-d7c9a8bb-cda5-4691-be82-6d7162c5b454'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-fa90a3e3-630e-4f40-bb07-dbbd2420e09c'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-9542f8c3-3ac5-4226-ac55-5f4f2b162021'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-f1168faa-9969-4dac-a115-c22e895abf10'\n",
      "distributed.core - ERROR - Exception while handling op heartbeat_worker\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/core.py\", line 493, in handle_comm\n",
      "    result = handler(comm, **msg)\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2196, in heartbeat_worker\n",
      "    ws._executing = {\n",
      "  File \"/home/ntolley/anaconda3/envs/sbi/lib/python3.8/site-packages/distributed/scheduler.py\", line 2197, in <dictcomp>\n",
      "    self.tasks[key]: duration for key, duration in executing.items()\n",
      "KeyError: 'run_simulator-2df1d8ec-bd46-4f0d-bb1a-5fa44e877a42'\n"
     ]
    }
   ],
   "source": [
    "client.cluster.scale(1000)\n",
    "def batch(seq, theta_samples, params_fname, prior_dict):\n",
    "    res_list= []\n",
    "    for sim_idx in seq:\n",
    "        res = dask.delayed(run_simulator)(theta_samples[sim_idx,:], params_fname, prior_dict, sim_idx)\n",
    "        res_list.append(res)\n",
    "        \n",
    "        \n",
    "\n",
    "    final_res = dask.compute(*res_list)\n",
    "    dpl_list = np.stack([final_res[idx][0] for idx in range(len(seq))])\n",
    "    \n",
    "    dpl_name = save_path + 'data/dpl_' + save_suffix + '_sim{}-{}'.format(seq[0],seq[-1]) + '.csv'\n",
    "    param_name = save_path + 'data/theta_' + save_suffix + '_sim{}-{}'.format(seq[0],seq[-1]) + '.csv'\n",
    "\n",
    "    np.savetxt(dpl_name, dpl_list, delimiter=',')\n",
    "    np.savetxt(param_name, theta_samples[seq,:].detach().cpu().numpy(), delimiter=',')\n",
    "    \n",
    "    dill_save([final_res[idx][1] for idx in range(len(seq))], 'data/spike_times', save_suffix + '_sim{}-{}'.format(seq[0],seq[-1]), save_path)\n",
    "    dill_save([final_res[idx][2] for idx in range(len(seq))], 'data/spike_gids', save_suffix + '_sim{}-{}'.format(seq[0],seq[-1]), save_path)\n",
    "    dill_save([final_res[idx][3] for idx in range(len(seq))], 'data/spike_types', save_suffix + '_sim{}-{}'.format(seq[0],seq[-1]), save_path)\n",
    "\n",
    "batches = []\n",
    "step_size = 1000\n",
    "for i in range(0, num_simulations, step_size):\n",
    "    print(i)\n",
    "    batch(list(range(i, i + step_size)),theta_samples, params_fname, prior_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (sbi)",
   "language": "python",
   "name": "sbi"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
